Export and import feature for WebApps

JW-CodeArt opened the pull request:
This PR adds a graphical and user-friendly import and export feature. It exports the ".desktop" files and the custom icons to an archive, which can then be imported again. If the browser is missing, it takes the most similar alternative.

Issues related to the feature:
https://github.com/linuxmint/webapp-manager/issues/292
https://github.com/linuxmint/webapp-manager/issues/338
https://github.com/linuxmint/webapp-manager/issues/347
https://github.com/linuxmint/webapp-manager/issues/22
https://github.com/linuxmint/webapp-manager/issues/232

Note: I tried to include the profiles directories, but it was unstable. For example, Firefox couldn't use a profile directory created with a newer version. 

There are new menu buttons:
![menu](https://github.com/user-attachments/assets/5cefcbf8-359f-4788-a17d-e0187286178f)

The new shortcuts are also documented in the shortcuts window:
![shortcuts](https://github.com/user-attachments/assets/a4aa5967-28a1-41c2-94ef-83c838835ed9)

After clicking the "Export" menu button, a file chooser dialog will appear:
![export_file_chooser](https://github.com/user-attachments/assets/f31a4e71-11f6-4322-b8a0-9083f4f3eb2a)

Once the process has finished, a new dialog will appear. Users can open the containing directory directly from the dialog.
![export_success_dialog](https://github.com/user-attachments/assets/93645a16-db64-4bcd-bbba-ebe3350cf213)
![import_success_dialog](https://github.com/user-attachments/assets/5a52e93d-e344-43b0-964e-3fa8bb70f209)

The archive contains two folders:
/applications/* with all the desktop files
/ice/icons/*  with all the icons
![archive](https://github.com/user-attachments/assets/f4ef907f-61bb-4cac-a9bb-aa032b873aca)

JW-CodeArt committed:
feature: "import/export" (1. iteration)
- import and export menu entry
- new shortcuts for the import and the export tool
- a new tool window (ei_tool.ui)
- import and export handler in common.py

JW-CodeArt committed:
feature "import/export" (2nd iteration)
- remove unnecessary code and simplify it
- remove the unstable profile directory export and import functionality
- remove the abort function and button, as the process is so quick that they are completely unnecessary.
- define gladefiles constants at the beginning of webapp-manager.py

JW-CodeArt committed:
feature "import/export" (3rd iteration)
- update text and function names
- removed the ei_task class as it is no longer needed
- created a get_codename() function as this is used twice in common.py
- removed the tool window and the progress tracking because they are not needed
- removed the browser profile export feature because it is unstable. It takes extremely long and does not work with different browser versions
- implemented a function which checks during the import process whether the browser is installed on the device. If not, it searches an alternative browser on the device.
- declared gladefiles paths as constants at the beginning of webapp-manager.py
- removed all trailing spaces
- applied the maximum line length

JW-CodeArt committed:
feature "import/export" (4th iteration)
- bug fix:
application crash
It now uses GLib.idle_add() to prevent the "Segmentation fault (core dumped)" error in GTK3
- try  catch in update_imported_desktop()

JW-CodeArt committed:
feature "import/export" (5th iteration)
- cleanup

JW-CodeArt committed:
merge conflict resolved

clefebvre commented:
Very nice. I like it, though the first thing that comes to mind, is if we really want to support this additional set of features.

Looking at the number of requests, we probably do, but I'd like it to be more minimalist.

- There's no need for keyboard shortcuts for import/export
- "Import" should be "Import..." to indicate a file path dialog follows. This is common practice for "Open...", "Save as..." etc.
- You could hardcode the path for the export and skip straight to the result dialog. ~/web-apps-YYY-MM-DD.tar.
- Do you handle collisions? What happens if you edit a webapp, reexport, reimport. Does it update it or create a duplicate?

JW-CodeArt commented:
Thanks for your feedback, Clefebvre. I will modify the code according your recommendations. 
Regards,
Jannik

JW-CodeArt committed:
feature "import/export" (6th iteration)
- revert unnecessary changes
- remove the keyboard shortcuts
- hardcoded the export path
- implemented better collision handling for re-export and re-import
- improved the browser check functionality when importing
- implemented draft for Suggestion:
Store custom icons automatically in the /ice/icons/ directory

JW-CodeArt committed:
resolved merge conflict

JW-CodeArt committed:
feature "import/export" (7th iteration)
-minor improvement to the success handler (the state variable type has been changed)
-updated some comments
